api/filter/http_connection_manager.proto

```diff
@@ -0,0 +1,250 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter;
+
+import "api/cds.proto";
+import "api/rds.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+message Rds {
+  // The name of an upstream cluster that hosts the route discovery service. The
+  // cluster must run a service that implements the RDS API. NOTE: This is the
+  // name of a cluster defined in the cluster manager configuration, not the
+  // full definition of a cluster as in the case of SDS and CDS.
+  // TODO(htuch): Should this be simplified to an UnresolvedAddress or the like
+  // in v2 API?
+  string cluster = 1;
+
+  // The name of the route configuration. This name will be passed to the RDS
+  // API. This allows an Envoy configuration with multiple HTTP listeners (and
+  // associated HTTP connection manager filters) to use different route
+  // configurations.
+  string router_config_name = 2;
+
+  // The delay, in milliseconds, between fetches to the RDS API. Envoy will add
+  // an additional random jitter to the delay that is between zero and
+  // refresh_delay_ms milliseconds. Thus the longest possible refresh delay is 2
+  // * refresh_delay_ms. Default value is 30000ms (30 seconds).
+  google.protobuf.Duration refresh_delay = 3;
+}
+
+// Filters on HTTP response/status code.
+message StatusCodeFilter {
+  // Comparison operator.
+  enum Op {
+    EQ = 0;
+    GE = 1;
+  }
+  Op op = 1;
+
+  // Default value to compare against if runtime value is not available.
+  google.protobuf.UInt32Value value = 2;
+
+  // Runtime key to get value for comparision. This value is used if defined.
+  string runtime_key = 3;
+}
+
+// Filters on total request duration.
+message DurationFilter {
+  // Comparison operator.
+  enum Op {
+    EQ = 0;
+    GE = 1;
+  }
+  Op op = 1;
+
+  // Default value to compare against if runtime value is not available.
+  google.protobuf.Duration value = 2;
+
+  // Runtime key to get value for comparision. This value is used if defined.
+  string runtime_key = 3;
+}
+
+// Filters for requests that are not health check requests. A health check
+// request is marked by the health check filter.
+message NotHealthCheckFilter {
+}
+
+// Filters for requests that are traceable. See the tracing overview for more
+// information on how a request becomes traceable.
+message TraceableFilter {
+}
+
+// Filters for random sampling of requests. Sampling pivots on the header
+// x-request-id being present. If x-request-id is present, the filter will
+// consistently sample across multiple hosts based on the runtime key value and
+// the value extracted from x-request-id. If it is missing, the filter will
+// randomly sample based on the runtime key value.
+message RuntimeFilter {
+  // Runtime key to get the percentage of requests to be sampled. This runtime
+  // control is specified in the range 0-100 and defaults to 0.
+  string key = 1;
```

Comment:
I know it matches the existing documentation but I find the usage of the word "key" awkward here.  It's basically the sample rate, right?